src/shell/shell_cmds.cpp

```diff
@@ -1409,33 +1498,37 @@ void DOS_Shell::CMD_DATE(char * args) {
 	const char* datestring = MSG_Get("SHELL_CMD_DATE_DAYS");
 	Bit32u length;
 	char day[6] = {0};
-	if (sscanf(datestring,"%u",&length) && (length<5) && (strlen(datestring) == (length*7+1))) {
+	if (sscanf(datestring, "%u", &length) && (length < 5) &&
+	    (strlen(datestring) == (length * 7 + 1))) {
 		// date string appears valid
-		for (Bit32u i = 0; i < length; i++) day[i] = datestring[reg_al*length+1+i];
+		for (Bit32u i = 0; i < length; i++)
+			day[i] = datestring[reg_al * length + 1 + i];
 	}
-	bool dateonly = ScanCMDBool(args,"T");
-	if (!dateonly) WriteOut(MSG_Get("SHELL_CMD_DATE_NOW"));
-
-	const char* formatstring = MSG_Get("SHELL_CMD_DATE_FORMAT");
-	if (strlen(formatstring)!=5) return;
-	char buffer[15] = {0};
-	Bitu bufferptr=0;
-	for (Bitu i = 0; i < 5; i++) {
-		if (i == 1 || i == 3) {
-			buffer[bufferptr] = formatstring[i];
-			bufferptr++;
-		} else {
-			if (formatstring[i] == 'M') bufferptr += sprintf(buffer+bufferptr,"%02u",(Bit8u) reg_dh);
-			if (formatstring[i] == 'D') bufferptr += sprintf(buffer+bufferptr,"%02u",(Bit8u) reg_dl);
-			if (formatstring[i] == 'Y') bufferptr += sprintf(buffer+bufferptr,"%04u",(Bit16u) reg_cx);
-		}
+	bool dateonly = ScanCMDBool(args, "T");
+	if (!dateonly) {
+		WriteOut(MSG_Get("SHELL_CMD_DATE_NOW"));
+		WriteOut("%s ", day);
+	}
+	WriteOut("%s\n",
+	         format_date((uint16_t)reg_cx, (uint8_t)reg_dh, (uint8_t)reg_dl));
+	if (!dateonly) {
+		WriteOut(MSG_Get("SHELL_CMD_DATE_SETHLP"), format);
 	}
-	WriteOut("%s %s\n",day, buffer);
-	if (!dateonly) WriteOut(MSG_Get("SHELL_CMD_DATE_SETHLP"));
 }
 
 void DOS_Shell::CMD_TIME(char * args) {
-	HELP("TIME");
+	char format[9], example[9];
+	char time_separator = dos.tables.country[13];
+	sprintf(format, "hh%cmm%css", time_separator,
+	        time_separator);
+	sprintf(example, "%u%c%02u%c%02u", 13, time_separator,
+	        14, time_separator, 15);
```

Comment:
Magic 13, 14, 15?

Reply:
The numbers 13, 14, 15 are just examples. I have put them in the string literal instead.